rir/src/compiler/opt/cleanup_checkpoints.cpp

```diff
@@ -0,0 +1,35 @@
+#include "../pir/pir_impl.h"
+#include "../transform/bb.h"
+#include "../transform/replace.h"
+#include "../util/cfg.h"
+#include "../util/visitor.h"
+#include "pass_definitions.h"
+
+namespace rir {
+namespace pir {
+
+void CleanupCheckpoints::apply(RirCompiler&, Closure* function) const {
+    auto apply = [](Code* code) {
+        std::unordered_set<BB*> toDelete;
+        Visitor::run(code->entry, [&](BB* bb) {
+            if (bb->isEmpty())
+                return;
+            if (auto cp = Checkpoint::Cast(bb->last())) {
```

Comment:
I split them into 3 passes since I want to be able to run them independently. Cleanup is run a couple of times, the other two are only run once. In general, at some point we remove checkpoints and then run the optimizations again without them, since that might give us more opportunities to optimize.